resolwe_bio/processes/variant_calling/gatk_variant_refinement.py

```diff
@@ -0,0 +1,126 @@
+    process_type = "data:variants:variants:refinevariants"
+    version = "1.0.0"
+    scheduling_class = SchedulingClass.BATCH
+    entity = {"type": "sample"}
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {
+            "docker": {"image": "public.ecr.aws/genialis/resolwebio/dnaseq:6.1.0"}
+        },
+        "resources": {
+            "cores": 2,
+            "memory": 16384,
+        },
+    }
+    data_name = '{{ vcf|sample_name|default("?") }}'
+
+    class Input:
+        """Input fields for GatkGenotypeRefinement."""
+
+        vcf_main = DataField(
+            "variants:vcf", label="The main input, as produced in the gatk-vqsr process"
+        )
+        ref_seq = DataField("seq:nucleotide", label="Reference sequence")
+
+        vcf_pop = DataField(
+            "variants:vcf",
+            label="Population-level variant set VCF data",
+            required=False,
+        )
+
+        ped = FileField(label="PED file describing pedigree", required=False)
+
+    class Output:
+        """Output fields for GatkGenotypeRefinement."""
+
+        vcf = FileField(label="Refined multi-sample vcf")
+        tbi = FileField(label="Tabix index")
+        species = StringField(label="Species")
+        build = StringField(label="Build")
+
+    def run(self, inputs, outputs):
+        """Run analysis."""
+
+        TMPDIR = os.environ.get("TMPDIR")
+
+        variants_cgp = "variants_recal_genotype_posteriors.vcf.gz"
+        variants_cgp_index = variants_cgp + ".tbi"
+        variants_refined = "variants.refined.vcf.gz"
+        variants_refined_index = variants_refined + ".tbi"
+
+        args_cgp = [
+            "-R",
+            inputs.ref_seq.output.fasta.path,
+            "-V",
+            inputs.vcf_main.output.vcf.path,
+            "-O",
+            variants_cgp,
+            "--tmp-dir",
+            TMPDIR,
+        ]
+
+        if inputs.vcf_pop:
+            args_cgp.extend(["--supporting", inputs.vcf_pop.output.vcf.path])
+        elif inputs.ped:
+            args_cgp.extend(["-ped", inputs.ped.import_file])
+
+        return_code, _, _ = Cmd["gatk"]["CalculateGenotypePosteriors"][args_cgp] & TEE(
+            retcode=None
+        )
+        if return_code:
+            self.error("GATK CalculateGenotypePosteriors failed.")
+
+        Cmd["tabix"]["-p", "vcf", "-f", variants_cgp]()
+
+        args_filtration = [
+            "-R",
+            inputs.ref_seq.output.fasta.path,
+            "-V",
+            variants_cgp,
+            "-O",
+            variants_refined,
+            "--genotype-filter-name",
+            "lowGQ",
+            "--genotype-filter-expression",
+            "GQ < 20",
+        ]
+
+        return_code, _, _ = Cmd["gatk"]["VariantFiltration"][args_filtration] & TEE(
+            retcode=None
+        )
+        if return_code:
+            self.error("GATK VariantFiltration failed.")
+
+        Path(variants_cgp).unlink(missing_ok=True)
```

Comment:
The intermediate files that are not explicitly referenced to the matching output fields will be automatically deleted, so it is not needed to delete them manually. The manual deletion only makes sense if multiple additional steps of processing of this VCF would follow, and one would try to save the storage space needed for completion of this analysis.